proposals/model-management.md

```diff
@@ -0,0 +1,109 @@
+# Model Management
+
+## Problem
+
+This proposal is attempting to solve the following problems:
+
+* Slow model loading on cold-start
+* Storage of LoRA adapters generated from finetuning jobs
+* Dynamic LoRA adapter injection
+
+While working within the following constraints:
+
+* KubeAI should be able to be installed with a single command
+* KubeAI should not rely on external dependencies to provide basic functionality
+* KubeAI should be able to run on a local machine
+* KubeAI should be able to be run on clouds with heterogenous disk features (limited support for ReadOnlyMany)
+
+## Solution
+
+KubeAI should support configurable storage for both models and adapters.
+
+These settings should be configurable at the `kind: Model` level because the optimum caching technique is relative to the size and frequency-of-use of the model. For example: GCP Hyperdisk ML is relatively expensive compared to other options, however it can be highly performant especially when it comes to loading large models.
+
+```yaml
+kind: Model
+spec:
+  storageProfile:
+    model: HyperdiskML
+    adapters: Bucket
+```
+
+Storage profiles should be defined at the KubeAI config.yaml level:
+
+```yaml
+# Set on Model objects if .spec.storageProfile is undefined.
+defaultStorageProfiles:
+  model: None # None is a special storage mode where each server pulls the model from source.
+  adapters: Bucket
```

Comment:
Maybe adapters should be None too and then it pulls from HF?

I am thinking we have a default where there is no requirement for any buckets, not even locally within the cluster.

Reply:
I was thinking about the situation where we support finetuning (a job that KubeAI manages the produces LoRA adapters). I was thinking we would need a storage mechanism not just a retrieval strategy. However you made me think we could just push those back up to huggingface with the token they already configured.

Reply:
Updated